proposals/predict_ml_model.json

```diff
@@ -1,7 +1,7 @@
 {
-    "id": "predict_random_forest",
-    "summary": "Predict values based on a Random Forest model",
-    "description": "Applies a Random Forest machine learning model to an array and predict a value for it.",
+    "id": "predict_ml_model",
```

Comment:
`predict_ml_model` looks a bit weird to me, it reads like you will be predicting the model (which is a confusing statement), instead of letting the model predict classes.

Something like `predict_class`, `ml_model_predict` or even `ml_predict` would feel better.

Note that we for the array, text, date related processes, we also use a prefix based naming (`array_append`, `array_apply`, `date_shift`) instead of a postfix based naming.

Reply:
My aim was to align with `predict_curve` so that in docs they would be listed side by side. Unfortunately, we are not very consistent with prefixes/suffixes (array_apply / date_shift / load_collection / save_result)...

Is there any other case where predict_class / predict_probabilities could become useful without ML?

Reply:
The thing with the name `predict_class`  is that that process would not only work for ML classification, but also ML regression, so ideally, the term `class` should be avoided. Unless there is a good reason to have separate inference/predict processes for ML classification and ML regression, but as far as I know that's not the case.


> Is there any other case where predict_class / predict_probabilities could become useful without ML? That was the main reason I added ml_model in the first place...

To me, "predict" implies "machine learning", so having both "ml" and "predict" in the name is slightly redundant.
However, it might be better for discoverability and self-documenting reasons to keep that bit of redundancy.